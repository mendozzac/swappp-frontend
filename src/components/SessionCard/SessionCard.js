import { useNavigate } from "react-router";
import path from "../../path/path";

const SessionCard = ({ session }) => {
  const navigate = useNavigate();
  console.log("1", session);
  const onSession = (session) => {
    navigate(`${path.season}/${session.id}`);
  };

  return (
    <li className="session-card">
      <div className="session-card__element" onClick={() => onSession(session)}>
        <span>{session.day}</span>
      </div>
    </li>
  );
};

export default SessionCard;
