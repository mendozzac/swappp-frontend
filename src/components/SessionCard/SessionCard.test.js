import { screen, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import SessionCard from "./SessionCard";
import { MemoryRouter } from "react-router-dom";
import ReactTestRenderer from "react-test-renderer";

describe("Given a SessionCard component", () => {
  describe("When it is called", () => {
    test("Then it should render a card", () => {
      const session = { day: "10/10/21", exercises: [] };
      render(
        <BrowserRouter>
          <SessionCard session={session} />
        </BrowserRouter>
      );
      const newSession = screen.getByText("10/10/21");

      expect(newSession).toBeInTheDocument();
    });
  });
  describe("When it receives a session", () => {
    test("Then it should render a card with the session", () => {
      const newSession = { day: "10/10/21", exercises: [] };

      const sessionCard = ReactTestRenderer.create(
        <MemoryRouter>
          <SessionCard session={newSession} />
        </MemoryRouter>
      );

      expect(sessionCard.toJSON()).toMatchSnapshot();
    });
  });
});
