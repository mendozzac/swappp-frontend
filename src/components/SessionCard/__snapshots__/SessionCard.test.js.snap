// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<li
  className="session-card"
>
  <div
    className="session-card__element"
    onClick={[Function]}
  >
    <span>
      10/10/21
    </span>
  </div>
</li>
`;

exports[`Given a SessionCard component When it receives a session Then it should render a card with the session 1`] = `
<li
  className="session-card"
>
  <div
    className="session-card__element"
    onClick={[Function]}
  >
    <span>
      10/10/21
    </span>
  </div>
</li>
`;
